src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import core.basesyntax.models.Circle;
+import core.basesyntax.models.IsoscelesTrapezoid;
+import core.basesyntax.models.Rectangle;
+import core.basesyntax.models.RightTriangle;
+import core.basesyntax.models.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final String DEFAULT_COLOR = "rgb(255, 255, 255)";
+    public static final double DEFAULT_RADIUS = 10.0;
+    private final ColorSupplier colorSupplier;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
+
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        double squareSide = random.nextDouble();
+        double width = random.nextDouble();
+        double height = random.nextDouble();
+        double firstLeg = random.nextDouble();
+        double secondLeg = random.nextDouble();
+        double radius = random.nextDouble();
+        double bottom = random.nextDouble();
+        double top = random.nextDouble();
+        double trapSide = random.nextDouble();
```

Comment:
The random values for dimensions should be within a realistic range to ensure meaningful figures are generated. Consider setting a minimum and maximum value for these random numbers.